language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/ContractUtil.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.rxjava;
+
+import com.daml.ledger.javaapi.data.*;
+import com.daml.ledger.javaapi.data.codegen.Contract;
+import com.daml.ledger.javaapi.data.codegen.ContractCompanion;
+import com.daml.ledger.javaapi.data.codegen.InterfaceCompanion;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * This class contains utilities to decode a <code>CreatedEvent</code> and create a <code>
+ * TransactionFilter</code> by provider parties It can only be instantiated with a subtype of <code>
+ * ContractCompanion</code>
+ */
+public final class ContractUtil<Ct> {
+  private final FromCreatedEventFunc<CreatedEvent, Ct> fromCreatedEvent;
+
+  private final Filter filter;
+
+  private ContractUtil(FromCreatedEventFunc<CreatedEvent, Ct> fromCreatedEvent, Filter filter) {
+    this.fromCreatedEvent = fromCreatedEvent;
+    this.filter = filter;
+  }
+
+  static <Ct> ContractUtil<Ct> of(ContractCompanion<Ct, ?, ?> companion) {
+    Filter filter = new InclusiveFilter(Set.of(companion.TEMPLATE_ID), Collections.emptyMap());
+    return new ContractUtil<>(companion::fromCreatedEvent, filter);
+  }
+
+  static <Cid, View> ContractUtil<Contract<Cid, View>> of(
```

Comment:
I suggest doing at least the ACS part of #15191 in this PR, if leaving aside the transaction stream part, as a sanity check for the API.

Reply:
Sure, it is now using the new ACS method in IouMain